### PR TITLE
Make k8s plugin fields private

### PR DESCRIPTION
**Pull request by Warashi**
**What this PR does**:

- make all the fields of types below private
    - provider.Manifest
    - provider.ResourceKey
- move `findUpdatedWorkloads` and `findConfigsAndSecrets` to provider package
- Rename `findUpdatedWorkloads` to `FindSameManifests` because this function does not see whether the manifests are updated or not.
- implement `provider.FindContainerImages` and use it in `determineVersions`
    - to decouple `determineVersions` from provider.Manifest.Body

**Why we need it**:

To restrict updates of field values.

**Which issue(s) this PR fixes**:

Fixes #

**Does this PR introduce a user-facing change?**:

- **How are users affected by this change**:
- **Is this breaking change**:
- **How to migrate (if breaking change)**:

### CODE REVIEW

File: pkg/app/pipedv1/plugin/kubernetes/deployment/determine_test.go [hunk 1]
```diff
@@ -624,448 +579,14 @@ spec:
...
-				manifests = append(manifests, mustUnmarshalYAML[provider.Manifest](t, []byte(strings.TrimSpace(data))))
+				manifests = append(manifests, mustParseManifests(t, data)...)
 			}
 			got := findWorkloadManifests(manifests, tt.refs)
 			assert.ElementsMatch(t, tt.want, got)
 		})
 	}
 }
 
-func TestFindUpdatedWorkloads(t *testing.T) {
```
Comment (Warashi) [inline pkg/app/pipedv1/plugin/kubernetes/deployment/determine_test.go:635]: These tests are moved to the provider package.

File: pkg/app/pipedv1/plugin/kubernetes/deployment/determine_test.go [hunk 2]
```diff
@@ -234,25 +234,6 @@ spec:
...
-  template:
-    spec:
-      containers:
-      - name: nginx
-        image: 12345
-`,
-			},
-			want:    nil,
-			wantErr: true,
-		},
```
Comment (Warashi) [inline pkg/app/pipedv1/plugin/kubernetes/deployment/determine_test.go:255]: This test case is an irregular case, and it's not critical. So I removed it.

File: pkg/app/pipedv1/plugin/kubernetes/deployment/determine_test.go [hunk 3]
```diff
@@ -269,22 +250,6 @@ spec:
...
-metadata:
-  name: no-containers-deployment
-spec:
-  template:
-    spec:
-      containers: "invalid-containers-field"
-`,
-			},
-			wantErr: true,
-		},
```
Comment (Warashi) [inline pkg/app/pipedv1/plugin/kubernetes/deployment/determine_test.go:287]: This test case is an irregular case, and it's not critical. So I removed it.

File: pkg/app/pipedv1/plugin/kubernetes/provider/manifest.go [hunk 4]
```diff
@@ -24,20 +24,24 @@ import (
 
 // Manifest represents a Kubernetes resource manifest.
 type Manifest struct {
-	Key  ResourceKey
-	Body *unstructured.Unstructured
+	key  ResourceKey
+	body *unstructured.Unstructured
```
Comment (Warashi) [inline pkg/app/pipedv1/plugin/kubernetes/provider/manifest.go:28]: This is what I want to do.

File: pkg/app/pipedv1/plugin/kubernetes/provider/resource.go [hunk 5]
```diff
@@ -66,55 +66,71 @@ const (
...
 
 type ResourceKey struct {
-	APIVersion string
-	Kind       string
-	Namespace  string
-	Name       string
+	apiVersion string
+	kind       string
+	namespace  string
+	name       string
```
Comment (Warashi) [inline pkg/app/pipedv1/plugin/kubernetes/provider/resource.go:72]: This is also what I want to do.

**Review by ffjlabo — APPROVED** [review]
LGTM

**Review by t-kikuc — APPROVED** [review]

**Pull request merged**